syntax = "proto3";
import "google/protobuf/struct.proto";
import "google/protobuf/empty.proto";
package microcrud;
option go_package ="microcudcontracts";
service CrudEntity{

  rpc Create (EntityProtocol) returns (ResultService) {}
  rpc Update (EntityProtocol) returns (ResultService) {}
  rpc Delete (DeleteProtocol) returns (ResultService) {}
  rpc GetById (GetProtocol) returns (ResultService) {}
  rpc GetData(google.protobuf.Empty) returns (ResultService) {}

}

// The request message containing the user's name.
message EntityProtocol {
  google.protobuf.Struct data =2;
}

message DeleteProtocol {
    string id=1;
}

message GetProtocol {
    string id=1;
}

// The response message containing the greetings.
message ResultService {
  string message = 1;
  google.protobuf.Struct data =2;
}